packages/webapp/src/components/Tile/Tiles.jsx

```diff
@@ -0,0 +1,55 @@
+/*
+ *  Copyright 2023 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+import React, { useMemo } from 'react';
+import PropTypes from 'prop-types';
+import IconLabelTile from './IconLabelTile';
+import { tyleTypes } from './constants';
+import styles from './styles.module.scss';
+
+const tileComponents = {
+  [tyleTypes.ICON_LABEL]: (props) => <IconLabelTile {...props} />,
+};
+
+/**
+ * A component that places tiles so that the empty space is evenly distributed for any window sizes.
+ * Either "children" or "tileType" and "tileData" props are required.
+ * See packages/webapp/src/stories/Tile/Tiles.stories.jsx for examples.
+ */
+export default function Tiles({ children, tileType, tileData, formatTileData, ...props }) {
+  const tiles = useMemo(() => {
```

Comment:
Just wondering if you know when is best to use useMemo? I am still not 100% comfortable the best situations and would be interested in the decision here. 

I have only read this article : https://blog.bitsrc.io/when-using-usememo-is-a-really-bad-idea-a2bdeb909812